src/platform/P6/BLEManagerImpl.cpp

```diff
@@ -0,0 +1,918 @@
+        // is now fully provisioned.
+#if CHIP_DEVICE_CONFIG_CHIPOBLE_DISABLE_ADVERTISING_WHEN_PROVISIONED
+        if (ConfigurationMgr().IsFullyProvisioned())
+        {
+            ClearFlag(mFlags, kFlag_AdvertisingEnabled);
+            ChipLogProgress(DeviceLayer, "CHIPoBLE advertising disabled because device is fully provisioned");
+        }
+#endif // CHIP_DEVICE_CONFIG_CHIPOBLE_DISABLE_ADVERTISING_WHEN_PROVISIONED
+
+        // Force the advertising state to be refreshed to reflect new provisioning state.
+        mFlags.Set(Flags::kFlag_AdvertisingRefreshNeeded, true);
+
+        DriveBLEState();
+        break;
+
+    default:
+        break;
+    }
+}
+
+bool BLEManagerImpl::SubscribeCharacteristic(BLE_CONNECTION_OBJECT conId, const ChipBleUUID * svcId, const ChipBleUUID * charId)
+{
+    ChipLogProgress(DeviceLayer, "BLEManagerImpl::SubscribeCharacteristic() not supported");
+    return false;
+}
+
+bool BLEManagerImpl::UnsubscribeCharacteristic(BLE_CONNECTION_OBJECT conId, const ChipBleUUID * svcId, const ChipBleUUID * charId)
+{
+    ChipLogProgress(DeviceLayer, "BLEManagerImpl::UnsubscribeCharacteristic() not supported");
+    return false;
+}
+
+bool BLEManagerImpl::CloseConnection(BLE_CONNECTION_OBJECT conId)
+{
+    ChipLogProgress(DeviceLayer, "Closing BLE GATT connection (con %u)", conId);
+
+    // Initiate a GAP disconnect.
+    wiced_bt_gatt_status_t gatt_err = wiced_bt_gatt_disconnect((uint16_t) conId);
+    if (gatt_err != WICED_BT_GATT_SUCCESS)
+    {
+        ChipLogError(DeviceLayer, "wiced_bt_gatt_disconnect() failed: %ld", gatt_err);
+    }
+
+    return (gatt_err == WICED_BT_GATT_SUCCESS);
+}
+
+uint16_t BLEManagerImpl::GetMTU(BLE_CONNECTION_OBJECT conId) const
+{
+    CHIPoBLEConState * p_conn;
+
+    /* Check if target connection state exists. */
+    p_conn = BLEManagerImpl::sInstance.GetConnectionState(conId);
+
+    if (!p_conn)
+    {
+        return wiced_bt_cfg_settings.gatt_cfg.max_mtu_size;
+    }
+    else
+    {
+        return p_conn->Mtu;
+    }
+}
+
+bool BLEManagerImpl::SendIndication(BLE_CONNECTION_OBJECT conId, const ChipBleUUID * svcId, const ChipBleUUID * charId,
+                                    PacketBufferHandle data)
+{
+    CHIP_ERROR err                  = CHIP_NO_ERROR;
+    uint16_t dataLen                = data->DataLength();
+    wiced_bt_gatt_status_t gatt_err = WICED_BT_GATT_SUCCESS;
+    CHIPoBLEConState * conState     = GetConnectionState(conId);
+
+    VerifyOrExit(conState != NULL, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+#ifdef BLE_DEBUG
+    ChipLogDetail(DeviceLayer, "Sending indication for CHIPoBLE TX characteristic (con %u, len %u)", conId, dataLen);
+#endif
+
+    // Send a indication for the CHIPoBLE TX characteristic to the client containing the supplied data.
+    gatt_err = wiced_bt_gatt_send_indication((uint16_t) conId, HDLC_CHIP_SERVICE_CHAR_C2_VALUE, dataLen, data->Start());
```

Comment:
Indication works with the Python controller and Chip tool command line. However, it is problematic with the Android chip-tool app that is expecting the use of notifications. If i recal right, the Spec mentions it should use notifications.

Reply:
Initially spec 0.7 Ballot version mentions this "indications" and it is updated to use notifications now. Most of the vendors are still using indications in "SendIndication" implementation. we will look at this as part of next release update.